nncf/openvino/quantization/quantize_model_if_op.py

```diff
@@ -0,0 +1,226 @@
+    """
+    Inserts extra outputs on If node inputs.
+
+    :param model_transformer: ModelTransformer instance.
+    :param model: Model instance.
+    :param if_node: If node.
+    :return: Model with extra outputs before If node.
+    """
+    transformation_layout = TransformationLayout()
+    for command in OVBackend.create_output_insertion_commands(model, if_node):
+        transformation_layout.register(command)
+    return model_transformer.transform(transformation_layout)
+
+
+def apply_algorithm_if_bodies(
+    algorithm: Algorithm,
+    parent_model: ov.Model,
+    parent_graph: NNCFGraph,
+    parent_dataset: Dataset,
+    subset_size: int,
+    parent_statistic_points: Optional[StatisticPointsContainer] = None,
+) -> ov.Model:
+    """
+
+    :param parent_model:
+    :param parent_graph:
+    :param parent_dataset:
+    :param parent_statistic_points:
+    :param parent_model_cnt:
+    :return:
+    """
+    nncf_logger.info(f"Quantize a new submodel")
+    quantized_model = algorithm.apply(parent_model, parent_graph, parent_statistic_points, parent_dataset)
+    if not has_if_op(parent_model):
+        return quantized_model
+    model_transformer_fp32 = factory.ModelTransformerFactory.create(parent_model)
+    for if_node in parent_graph.get_nodes_by_metatypes([OVBackend.if_node_metatype()]):
+        then_model_input_names = OVBackend.get_if_subgraph_input_names(parent_model, if_node, True)
+        else_model_input_names = OVBackend.get_if_subgraph_input_names(parent_model, if_node, False)
+        if_cond_input_name = OVBackend.get_if_cond_input_name(parent_model, if_node)
+        parent_model_with_additional_outputs = _add_outputs_before_if_node(
+            model_transformer_fp32, parent_model, if_node
+        )
+        then_dataset, else_dataset = _make_dataset_for_children_model(
+            factory.EngineFactory.create(parent_model_with_additional_outputs),
+            parent_dataset,
+            if_cond_input_name,
+            then_model_input_names,
+            else_model_input_names,
+            subset_size,
+        )
+        then_model = _extract_if_submodel(model_transformer_fp32, if_node, True)
+        else_model = _extract_if_submodel(model_transformer_fp32, if_node, False)
+        then_quantized_model = apply_algorithm_if_bodies(
+            algorithm, then_model, NNCFGraphFactory.create(then_model), then_dataset, subset_size, None
+        )
+        else_quantized_model = apply_algorithm_if_bodies(
+            algorithm, else_model, NNCFGraphFactory.create(else_model), else_dataset, subset_size, None
+        )
+        model_transformer_int8 = factory.ModelTransformerFactory.create(quantized_model)
+        quantized_model = _update_if_submodel(model_transformer_int8, if_node, True, then_quantized_model)
+        model_transformer_int8 = factory.ModelTransformerFactory.create(quantized_model)
```

Comment:
Why not use one instance of the `ModelTransformer` since we create it based on the main model?